docs/cogs.rst

```diff
@@ -0,0 +1,148 @@
+.. currentmodule:: nextcord
+
+.. _cogs:
+
+Cogs
+====
+
+There comes a point in your bot's development when you want to organize a collection of commands, listeners, and some state into one class. Cogs allow you to do just that.
+
+The gist:
+
+- Each cog is a Python class that subclasses :class:`nextcord.Cog`.
+- Every command is marked with the :func:`.slash_command`, :func:`.user_command`, or :func:`.message_command` decorators.
+- Every listener is marked with the :meth:`nextcord.Cog.listener` decorator.
+- Cogs are then registered with the :meth:`.Client.add_cog` call.
+- Cogs are subsequently removed with the :meth:`.Client.remove_cog` call.
+
+Quick Example
+-------------
+
+This example cog defines a ``Greetings`` category for your commands, with a single :class:`nextcord.SlashApplicationCommand` named ``hello`` as well as a listener to listen to an :ref:`Event <discord-api-events>`.
+
+.. code-block:: python3
+
+    class Greetings(nextcord.Cog):
+        def __init__(self, client):
+            self.client = client
+            self._last_member = None
+
+        @nextcord.Cog.listener()
+        async def on_member_join(self, member):
+            channel = member.guild.system_channel
+            if channel is not None:
+                await channel.send(f"Welcome {member.mention}.")
+
+        @nextcord.slash_command()
+        async def hello(self, interaction, member: nextcord.Member = None):
+            """Says hello"""
+            member = member or interaction.user
+            if self._last_member is None or self._last_member.id != member.id:
+                await interaction.response.send_message(f"Hello {member.name}~")
+            else:
+                await interaction.response.send_message(f"Hello {member.name}... This feels familiar.")
+            self._last_member = member
+
+A couple of technical notes to take into consideration:
+
+- All listeners must be explicitly marked via the :meth:`nextcord.Cog.listener` decorator.
+- The name of the cog is automatically derived from the class name but can be overridden. See :ref:`cogs_meta_options`.
+- All commands must now take a ``self`` parameter to allow usage of instance attributes that can be used to maintain state.
```

Comment:
Do we want to add the setup function to this as a kind of "full example"?

Reply:
We could, but the problem is extensions are only in `ext.commands.Bot` and they haven't been implemented in `nextcord.Client` at all